pints/_mcmc/_monomial_gamma_hmc.py

```diff
@@ -0,0 +1,507 @@
+#
+# Monomial gamma HMC MCMC method
+#
+# This file is part of PINTS.
+#  Copyright (c) 2017-2019, University of Oxford.
+#  For licensing information, see the LICENSE file distributed with the PINTS
+#  software package.
+#
+from __future__ import absolute_import, division
+from __future__ import print_function, unicode_literals
+import pints
+import numpy as np
+from scipy import integrate
+from scipy import interpolate
+
+
+class MonomialGammaHMCMCMC(pints.SingleChainMCMC):
+    """
+    *Extends:* :class:`SingleChainMCMC`
+
+    Implements Monomial Gamma HMC as described in [1] - a generalisation
+    of HMC as described in [2] - involving a non-physical kinetic energy term.
+
+    Uses a physical analogy of a particle moving across a landscape under
+    Hamiltonian dynamics to aid efficient exploration of parameter space.
+    Introduces an auxilary variable -- the momentum (``p_i``) of a particle
+    moving in dimension ``i`` of negative log posterior space -- which
+    supplements the position (``q_i``) of the particle in parameter space. The
+    particle's motion is dictated by solutions to Hamilton's equations,
+
+        dq_i/dt =   partial_d H/partial_d p_i,
```

Comment:
Please wrap in math block https://github.com/pints-team/pints/blob/master/CONTRIBUTING.md#using-maths-in-documentation

Reply:
Done -- thanks for the typesetting ref here. Useful!